src/promptflow/promptflow/executor/_service/apis/execution.py

```diff
@@ -5,52 +5,95 @@
+    request_id = OperationContext.get_instance().request_id
+    executor_version = get_executor_version()
+    with get_service_log_context(node_request):
+        service_logger.info(
+            f"Received node execution request, node name: {node_request.node_name}, "
+            f"request id: {request_id}, executor version: {executor_version}."
+        )
+        try:
+            result = await invoke_function_in_process(node_request, request.headers, single_node_run)
+            service_logger.info(f"Completed node execution request, node name: {node_request.node_name}.")
+            return result
+        except Exception as ex:
+            error_type_and_message = (f"({ex.__class__.__name__}) {ex}",)
+            service_logger.error(
+                f"Failed to execute node, node name: {node_request.node_name}. Error: {error_type_and_message}"
+            )
+            raise ex
+
+
+async def flow_test(flow_request: FlowExecutionRequest):
     # validate request
     flow_request.validate_request()
     # resolve environment variables
     set_environment_variables(flow_request)
     # execute flow
     storage = DefaultRunStorage(base_dir=flow_request.working_dir, sub_dir=flow_request.output_dir)
-    with get_log_context(flow_request):
-        return execute_flow(
-            flow_request.flow_file,
-            flow_request.working_dir,
-            flow_request.output_dir,
-            flow_request.connections,
-            flow_request.inputs,
-            run_id=flow_request.run_id,
-            storage=storage,
-        )
+    return execute_flow(
+        flow_request.flow_file,
+        flow_request.working_dir,
+        flow_request.output_dir,
+        flow_request.connections,
+        flow_request.inputs,
+        run_id=flow_request.run_id,
+        storage=storage,
+    )
 
 
-@router.post("/execution/node")
-async def node_execution(request: Request, node_request: NodeExecutionRequest):
-    OperationContext.get_instance().update(dict(request.headers))
+async def single_node_run(node_request: NodeExecutionRequest):
     # validate request
     node_request.validate_request()
     # resolve environment variables
     set_environment_variables(node_request)
-    # execute node
-    with get_log_context(node_request):
-        storage = DefaultRunStorage(base_dir=node_request.working_dir, sub_dir=node_request.output_dir)
-        result = FlowExecutor.load_and_exec_node(
-            node_request.flow_file,
-            node_request.node_name,
-            flow_inputs=node_request.flow_inputs,
-            dependency_nodes_outputs=node_request.dependency_nodes_outputs,
-            connections=node_request.connections,
-            working_dir=node_request.working_dir,
-            storage=storage,
-        )
-        return result
+    storage = DefaultRunStorage(base_dir=node_request.working_dir, sub_dir=node_request.output_dir)
+    return FlowExecutor.load_and_exec_node(
```

Comment:
Please do not directly invoke a long running sync call in a async function.

Reply:
Updated in another PR.
https://github.com/microsoft/promptflow/pull/2043/